projects/plugins/boost/compatibility/yoast.php

```diff
@@ -0,0 +1,12 @@
+<?php
+/**
+ * Yoast SEO compatibility for Boost
+ *
+ * @package automattic/jetpack-boost
+ */
+
+namespace Automattic\Jetpack_Boost\Compatibility\Yoast;
+
+// Add the Critical CSS generation query arg to Yoast's allowed query args list.
+// This prevents Yoast from removing the query arg, which breaks Critical CSS generation.
+add_filter( 'Yoast\WP\SEO\allowlist_permalink_vars', array( '\Automattic\Jetpack_Boost\Modules\Optimizations\Critical_CSS\Critical_CSS', 'add_critical_css_query_arg_to_list' ) );
```

Comment:
I assume `query_vars` did not work, and Yoast relies on their own filter.

However, I think we should still look into using the filter alongside this one. This is because other less popular plugins may still rely on `query_vars` as it is a WP standard.

Reply:
`query_vars` isn't used to make this work in the context of both plugins - they both rely on custom logic and filters. While reading your comment, initially I also thought that we should use the filter. However, after several re-reads, I figured out that with the word "registered", they just mean registered in the context of the plugin - via UI or code, no by query_args.

I also checked out the code for any references of `query_vars` in both plugins and they don't use it at all in the context of query arg removal.

Reply:
After reading on query vars, I felt it was unnecessary to add it (since we're not working with WP_Query or using it for pretty permalinks).

However since WordPress is used in ways we can't predict, I've added the token to the WP query vars list (only if the get param is present). This also affects the filters of the plugins, as they use the updated method.